src/ascii_str.rs

```diff
@@ -57,60 +61,19 @@ impl AsciiStr {
         self.as_mut_slice().as_mut_ptr()
     }
 
+    #[cfg(not(feature = "no_std"))]
     /// Copies the content of this `AsciiStr` into an owned `AsciiString`.
     pub fn to_ascii_string(&self) -> AsciiString {
         AsciiString::from(self.slice.to_vec())
     }
 
-    /// Converts anything that can represent a byte slice into an `AsciiStr`.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// # use ascii::AsciiStr;
-    /// let foo = AsciiStr::from_bytes("foo");
-    /// let err = AsciiStr::from_bytes("Ŋ");
-    /// assert_eq!(foo.unwrap().as_str(), "foo");
-    /// assert_eq!(err, Err(()));
-    /// ```
-    pub fn from_bytes<'a, B: ?Sized>(bytes: &'a B) -> Result<&'a AsciiStr, ()>
-        where B: AsRef<[u8]>
-    {
-        unsafe {
-            if bytes.as_ref().is_ascii() {
-                Ok( Self::from_bytes_unchecked(bytes) )
-            } else {
-                Err(())
-            }
-        }
-    }
-
-    /// Converts anything that can be represented as a byte slice to an `AsciiStr` without checking for non-ASCII characters..
-    ///
-    /// # Examples
-    /// ```
-    /// # use ascii::AsciiStr;
-    /// let foo = unsafe{ AsciiStr::from_bytes_unchecked("foo") };
-    /// assert_eq!(foo.as_str(), "foo");
-    /// ```
-    pub unsafe fn from_bytes_unchecked<'a, B: ?Sized>(bytes: &'a B) -> &'a AsciiStr
-        where B: AsRef<[u8]>
-    {
-        mem::transmute(bytes.as_ref())
-    }
-
-    /// Converts a borrowed string to a borrows ascii string.
-    pub fn from_str<'a>(s: &'a str) -> Result<&'a AsciiStr, ()> {
-        AsciiStr::from_bytes(s.as_bytes())
-    }
-
```

Comment:
I guess these methods are replaced with the methods of the `AsAsciiStr` and `AsMutAsciiStr` traits? I'd like to keep the inherent functions `from_bytes` and `from_bytes_unchecked`, they can forward to the trait implementations. Since `from_bytes` is generic enough we can drop `from_str`.

The reason is, that `str` also provides a non-generic construction via `str::from_utf8` and `String` has `String::from`. If you don't need generic conversions at function boundaries you also don't need to use the conversion traits, only `use AsciiStr`.

What do you think about it?